tests/app/celery/test_tasks.py

```diff
@@ -74,3 +77,24 @@ def test_should_not_send_sms_if_db_peristance_failed(sample_template, mocker):
     with pytest.raises(NoResultFound) as e:
         notifications_dao.get_notification(sample_template.service_id, notification_id)
     assert 'No row was found for one' in str(e.value)
+
+
+def test_should_send_sms_code(mocker):
+    notification = {'to': '+441234123123',
+                    'secret_code': '12345'}
+
+    encrypted_notification = encryption.encrypt(notification)
+
+    mocker.patch('app.firetext_client.send_sms')
+    send_sms_code(encrypted_notification)
+    firetext_client.send_sms.assert_called_once_with(notification['to'], notification['secret_code'])
+
+
+def test_should_log_firetext_client_exception(mocker):
+    notification = {'to': '+441234123123',
+                    'secret_code': '12345'}
+
+    encrypted_notification = encryption.encrypt(notification)
+    mocker.patch('app.firetext_client.send_sms', side_effect=FiretextClientException)
+    send_sms_code(encrypted_notification)
+    firetext_client.send_sms.assert_called_once_with(notification['to'], notification['secret_code'])
```

Comment:
I didn't actually know what how to test that the logging happened. If there is a failure we might want to put it in a retry queue